core/src/main/java/org/eclipse/hono/util/TenantConstants.java

```diff
@@ -66,6 +66,12 @@
      */
     public static final String FIELD_AUTO_PROVISIONING_ENABLED = "auto-provisioning-enabled";
 
+    /**
+     * The name of the property that indicates whether a unregistered device that authenticates with
+     * a client certificate should be auto-provisioned as a gateway. 
+     */
+    public static final String FIELD_AUTO_PROVISION_AS_GATEWAY = "auto-provision-as-gateway";
+
```

Comment:
this is not needed anymore, or is it?

Reply:
This is currently used in the test classes `AbstractTenantServiceTest` and `FileBasedTenantServiceTest`.

If we don't want to this property to showup in `TenantObject` then we could remove that while mapping to `TenantObject` using `DeviceRegistryUtils.convertTenant()`. But I don't see any drawbacks why it should be removed.

Reply:
IMHO we should *not* include the property in the TenantObject returned to protocol adapters because this is information that is *internal* to the registry and is of no concern to the adapters. This way we also prevent adapters to *suddenly* depend on this information ...

Reply:
Now this property is not included in the `TenantObject` anymore. I have pushed the changes for it.